Client/Pages/BookPages/BookView.razor

```diff
@@ -1,97 +1,130 @@
-        {
-            NavigationManager.NavigateTo("/publisherView/" + id);
-        }
-
-        private void Details(Guid? id)
-        {
-            NavigationManager.NavigateTo("/bookview/" + id);
-        }
+        private List<Test.Shared.Book> books = new List<Test.Shared.Book>();
+        private List<Author> authors { get; set; } = new List<Author>();
+        private Test.Shared.Book _book = new Test.Shared.Book();
+        private List<Guid> AuthorIds { get; set; } = new List<Guid>();
 
         private async Task AddBook()
         {
-            Test.Shared.Book book = new Test.Shared.Book { BookName = bookName, BookPublisherId = publisherId };
-            Test.Shared.Publisher publisher = await Http.GetFromJsonAsync<Test.Shared.Publisher>($"publisher/{book.BookPublisherId}");
-            await Http.PostAsJsonAsync<Test.Shared.Book>("book", book);
-            StateContainer.books = await Http.GetFromJsonAsync<List<Test.Shared.Book>>("book");
-            bookName = "";
-            publisherId = new Guid();
+            var responce = await Http.PostAsJsonAsync<Test.Shared.Book>("api/books", _book);
+            var book = responce.Content.ReadAsAsync<Book>().Result;
+            foreach (Guid authorId in AuthorIds)
+            {
+                BookAuthor bookAuthor = new BookAuthor
+                {
+                    BookId = book.BookId,
+                    AuthorId = authorId
+                };
+                await Http.PostAsJsonAsync<BookAuthor>("api/bookauthors", bookAuthor);
+            }
+            _book = new Test.Shared.Book();
+            books = await Http.GetFromJsonAsync<List<Test.Shared.Book>>("api/books");
         }
+
         private async Task Delete(Guid? id)
         {
-            Console.WriteLine("deleting?");
-            await Http.DeleteAsync($"book/{id}");
-            StateContainer.books = await Http.GetFromJsonAsync<List<Test.Shared.Book>>("book");
+            await Http.DeleteAsync($"api/books/{id}");
+            books = await Http.GetFromJsonAsync<List<Test.Shared.Book>>("api/books");
         }
 
         protected override async Task OnInitializedAsync()
         {
-            StateContainer.books = await Http.GetFromJsonAsync<List<Test.Shared.Book>>("book");
-            StateContainer.OnChange += StateHasChanged;
+            books = await Http.GetFromJsonAsync<List<Test.Shared.Book>>("api/books");
+            publishers = await Http.GetFromJsonAsync<List<Test.Shared.Publisher>>("api/publishers");
+            authors = await Http.GetFromJsonAsync<List<Author>>("api/authors");
         }
 
-        public void Dispose()
+        private void CheckBoxClicked(Guid id, object checkedValue)
         {
-            StateContainer.OnChange -= StateHasChanged;
+            Console.WriteLine("Am here " + id);
+            if ((bool)checkedValue)
+            {
+                if (!AuthorIds.Contains(id))
+                {
+                    AuthorIds.Add(id);
```

Comment:
I would prefer a one-liner here:
`if(!AuthorIds.Contains(id)) AuthorIds.Add(id);`

Using control structures without curly braces is discouraged. However the only place is actually one-liners, which dramatically improves readability.